### ISSUE TITLE
[Proposal] Make the MAX_RRD_SPEED configurable for users who need it larger than default

### ISSUE THREAD

**Issue by netravnen** [librenms/librenms#16231]
> Sure, send a change to update the value to `100000000000`, remember
> that it will only affect new RRDs.
> 
> _Originally posted by @murrant in https://github.com/librenms/librenms/issues/14582#issuecomment-1354093617_

**Proposal**

Make the `MAX_RRD_SPEED` configurable.

Allow the value to be in range of `-gt` (greater than) current default, and `-leq` (lower than or equal to) the current default times 24.

Example, lower bound `-gt` 100G, upper bound `-leq` (100G * 24)

**Addition**

If the default MAX_RRD_SPEED is upgraded. Present the users of LibreNMS with a high priority warning the setting in the local configuration is `-leq` (lower or equal) to the current default.

**When changing MAX_RRD_SPEED**

Warning: The user should run rrdtune to ensure the max port speed for rrd files is correct. (or be run automatically by the upgrade script)

**If local MAX_RRD_SPEED -lt current default**

Alert: Recommend/Push the users to remove the locally reconfigured MAX_RRD_SPEED value.

Warning: The user should run rrdtune to ensure the max port speed for rrd files is correct. (or be run automatically by the upgrade script)

**Comment by murrant** [comment]
Not much a fan of anything where the answer is to punt it to the users.

**Comment by netravnen** [comment]
@murrant You are right in that regard. It may not be the preferred solution/option/route to go.

Weighing in on the other hand. I think it is about time the MAX RRD speed value is raised beyond 100G. - If I read https://github.com/librenms/librenms/issues/14582#issuecomment-1354093617 correctly. A 200G BOND port would be initialised with a 100G max speed in the RRD file... 🤔 💭 

----

MT is arriving on the market with more and more [100G equipment](https://mikrotik.com/products?filter&s=c&search=XQ) as the months roll by.
Recent addition is [CRS520-4XS-16XQ-RM](https://mikrotik.com/product/crs520_4xs_16xq_rm) (16x100GbE, 4x25GbE)

Older Nvidia Mellanox SN2700 [(Datasheet)](https://network.nvidia.com/files/doc-2020/pb-sn2700.pdf) switches being available for a 'modest' ~€ 1K-1½K refurbished.

**Comment by netravnen** [comment]
Alternatively, include the mention of RRDtune in the reference configuration file. To make it obvious for users there exists an option to automatically adjust the max value of RRD files to the current max value of the interface (port).

```php
# RRDTune
#
# When we create rrd files for ports, we currently do so with a max value of 12500000000 
# (100G). Because of this if a device sends us bad data back then it can appear as 
# though a 100M port is doing 40G+ which is impossible. To counter this you can enable 
# the rrdtool tune option which will fix the max value to the interfaces physical speed 
# (minimum of 10M).
#
# Now when a port interface speed changes (this can happen because of a physical change 
# or just because the device has misreported) the max value is set.
$config['rrdtool_tune'] = true;
```

**Comment by murrant** [comment]
I would definitely accept a PR to increase the default value as 100G is becoming very common.

**Comment by laf** [comment]
Would it make sense to set the max port speed based on the discovered port speed? We can also change it when we detect a change in port speed (not sure how much that might flip flop though)

**Comment by murrant** [comment]
@laf we already do that if tune port (or some such) is set.
